community/kernel/src/test/java/org/neo4j/helpers/FakeClock.java

```diff
@@ -47,8 +50,31 @@ public long nanoTime()
         return time;
     }
 
-    public void forward( long amount, TimeUnit timeUnit )
+    public synchronized void forward( long amount, TimeUnit timeUnit )
```

Comment:
It was in ArtificialClock originally. Someone who designed that class likely thought it important.
But given it's called rarely (in a single test), and it is not in production code, I'd be inclined to leave it in.